packages/honeycomb-opentelemetry-web/examples/hello-world-react-create-app/src/reactRouterSpanProcessor.ts

```diff
@@ -0,0 +1,48 @@
+import { SpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { Span } from '@opentelemetry/api';
+
+/**
+ * SpanProcessor that adds attributes to spans based on the state of the React Router
+ * Sets the page.route attribute to the generic dynamic route
+ * Records the span as an error if there are errors in the router state (e.g. 404)
+ */
+export class ReactRouterSpanProcessor implements SpanProcessor {
+  router;
```

Comment:
[optional] as far as i can tell, we don't need to hang on to `router`; we use it in the constructor and then never again?

Reply:
We still have to define it here to be able to use `this.router` in the constructor!